src/confcom/azext_confcom/os_util.py

```diff
@@ -219,14 +270,27 @@ def map_image_from_tar(image_name: str, tar: TarFile, tar_location: str):
     image_info_file_path = os.path.join(tar_dir, info_file)
     image_info_raw = load_json_from_file(image_info_file_path)
     # delete the extracted json file to clean up
-    os.remove(image_info_file_path)
+    force_delete_silently(image_info_file_path)
     image_info = image_info_raw.get("config")
     # importing the constant from config.py gives a circular dependency error
     image_info["Architecture"] = image_info_raw.get("architecture")
 
     return image_info
 
 
+# sometimes image tarfiles have readonly members. this will try to change their permissions and delete them
+def force_delete_silently(filename: str) -> None:
+    try:
+        os.chmod(filename, stat.S_IWRITE)
+    except FileNotFoundError:
+        pass
```

Comment:
Should you have an error message here?

Reply:
The idea is the file may or may not be there. we only want to delete it if it's there. Since the other errors might happen, I figured it it's a little cleaner to do it this way